src/cdomain/value/cdomains/int/intervalDomain.ml

```diff
@@ -79,11 +80,77 @@ struct
   (* TODO: change to_int signature so it returns a big_int *)
   let to_int x = Option.bind x (IArith.to_int)
   let of_interval ?(suppress_ovwarn=false) ik (x,y) = norm ~suppress_ovwarn ik @@ Some (x,y)
+
+  let of_bitfield ik x = 
+    let min ik (z,o) = 
+      let signBit = Ints_t.shift_left Ints_t.one ((Size.bit ik) - 1) in 
+      let signMask = Ints_t.lognot (Ints_t.of_bigint (snd (Size.range ik))) in
+      let isNegative = Ints_t.logand signBit o <> Ints_t.zero in
+      if isSigned ik && isNegative then Ints_t.logor signMask (Ints_t.lognot z)
+      else Ints_t.lognot z
+    in let max ik (z,o) =
+         let signBit = Ints_t.shift_left Ints_t.one ((Size.bit ik) - 1) in 
+         let signMask = Ints_t.of_bigint (snd (Size.range ik)) in
+         let isPositive = Ints_t.logand signBit z <> Ints_t.zero in
+         if isSigned ik && isPositive then Ints_t.logand signMask o
+         else o 
+    in fst (norm ik (Some (min ik x, max ik x)))
+
   let of_int ik (x: int_t) = of_interval ik (x,x)
   let zero = Some IArith.zero
   let one  = Some IArith.one
   let top_bool = Some IArith.top_bool
 
+  let to_bitfield ik z = 
+    match z with None -> (Ints_t.lognot Ints_t.zero, Ints_t.lognot Ints_t.zero) | Some (x,y) ->
+      let (min_ik, max_ik) = Size.range ik in
+      let startv = Ints_t.max x (Ints_t.of_bigint min_ik) in
```

Comment:
Can you make this function easier to read?
A comment about what is done here overall may help. Possibly one may also find a more telling name for the helper function `analyze_bits`?